### PR TITLE
GH4030: Fix GitLab CI runner tags

### LINKED ISSUE #4030: GitLab CI runner tags are not split correctly
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

2.3.0

### Operating system

Linux

### Operating system architecture

64-Bit

### CI Server

GitLab CI

### What are you seeing?

I noticed GitLab CI runner tags are not split correctly while playing around with [gitfool/GitLabCI.Dungeon](https://gitlab.com/gitfool/GitLabCI.Dungeon), as shown this [job](https://gitlab.com/gitfool/GitLabCI.Dungeon/-/jobs/3246542880), the `CI_RUNNER_TAGS` environment variable contains brackets and quotes:
```
CI_RUNNER_TAGS=["gce", "east-c", "linux", "ruby", "mysql", "postgres", "mongo", "git-annex", "shared", "docker", "saas-linux-small-amd64"]
```
which is split into a string array that still has these brackets and quotes:
```
BuildSystem.GitLabCI.Environment.Runner.Tags[0]               ["gce"
BuildSystem.GitLabCI.Environment.Runner.Tags[1]               "east-c"
BuildSystem.GitLabCI.Environment.Runner.Tags[2]               "linux"
BuildSystem.GitLabCI.Environment.Runner.Tags[3]               "ruby"
BuildSystem.GitLabCI.Environment.Runner.Tags[4]               "mysql"
BuildSystem.GitLabCI.Environment.Runner.Tags[5]               "postgres"
BuildSystem.GitLabCI.Environment.Runner.Tags[6]               "mongo"
BuildSystem.GitLabCI.Environment.Runner.Tags[7]               "git-annex"
BuildSystem.GitLabCI.Environment.Runner.Tags[8]               "shared"
BuildSystem.GitLabCI.Environment.Runner.Tags[9]               "docker"
BuildSystem.GitLabCI.Environment.Runner.Tags[10]              "saas-linux-small-amd64"]
```

### What is expected?

The runner tags should be split into a string array that strips these brackets and quotes:
```
BuildSystem.GitLabCI.Environment.Runner.Tags[0]               gce
BuildSystem.GitLabCI.Environment.Runner.Tags[1]               east-c
BuildSystem.GitLabCI.Environment.Runner.Tags[2]               linux
BuildSystem.GitLabCI.Environment.Runner.Tags[3]               ruby
BuildSystem.GitLabCI.Environment.Runner.Tags[4]               mysql
BuildSystem.GitLabCI.Environment.Runner.Tags[5]               postgres
BuildSystem.GitLabCI.Environment.Runner.Tags[6]               mongo
BuildSystem.GitLabCI.Environment.Runner.Tags[7]               git-annex
BuildSystem.GitLabCI.Environment.Runner.Tags[8]               shared
BuildSystem.GitLabCI.Environment.Runner.Tags[9]               docker
BuildSystem.GitLabCI.Environment.Runner.Tags[10]              saas-linux-small-amd64
```

### Steps to Reproduce

N/A.

### Output log

_No response_

### PR DESCRIPTION
**Pull request by gitfool**
Fixes #4030.

### CODE REVIEW

File: src/Cake.Common/Build/GitLabCI/Data/GitLabCIRunnerInfo.cs [hunk 1]
```diff
@@ -47,10 +47,10 @@ public string[] Tags
         {
             get
             {
-                var tags = GetEnvironmentString("CI_RUNNER_TAGS").Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                var tags = GetEnvironmentString("CI_RUNNER_TAGS").Trim('[', ']').Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (devlead) [inline src/Cake.Common/Build/GitLabCI/Data/GitLabCIRunnerInfo.cs:50]: Question: could the variables contain a `,` within quotes?

Would we need to do something similar to the [QuoteAwareStringSplitter](https://github.com/cake-build/cake/blob/a1893a2bba26f49e089d66beeb78421cedf6126d/src/Cake.Core/Text/QuoteAwareStringSplitter.cs#L14)?

Comment (gitfool) [reply src/Cake.Common/Build/GitLabCI/Data/GitLabCIRunnerInfo.cs:50]: I did a test with a self-hosted instance and, although the runner ui let me create a tag with commas, it had to be quoted; but then when editing the tags again they were no longer quoted and got split into multiple tags. So I would argue that such tags are not expected and not really supported. This makes sense given the runner ui states: "You can set up jobs to only use runners with specific tags. Separate tags with commas."

Comment (devlead) [reply src/Cake.Common/Build/GitLabCI/Data/GitLabCIRunnerInfo.cs:50]: Ok, then we'll revisit if it becomes an issue

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@gitfool your changes have been merged, thanks for your contribution 👍

**Pull request merged**